src/screens/Asset/ReceiveModal.js

```diff
@@ -61,151 +61,96 @@ type StateProps = {|
-            labelStyle={{ fontSize: fontSizes.tiny }}
-            primary
-          />
-        ),
-      }] : undefined}
+      // TODO : The label itself will be revisited later on the direction of the product team
+      // headerLeftItems={showErc20Note ? [{
+      //   custom: (
+      //     <LabelBadge
+      //       label={t('label.erc20TokensOnly')}
+      //       labelStyle={{ fontSize: fontSizes.tiny }}
+      //       primary
+      //     />
+      //   ),
+      // }] : undefined}
       centerFloatingItem={
         <ImageWrapper style={{ position: 'absolute', marginTop: -24 }}>
           <ProfileImage userName={username} diameter={48} />
         </ImageWrapper>
       }
     >
       <ContentWrapper forceInset={{ top: 'never', bottom: 'always' }}>
-        <WarningBanner rounded small />
-        {!!ensName && (
-          <InfoView>
-            <BaseText
-              big
-              onPress={() => handleCopyToClipboard(ensName, true)}
-              center
-            >
+        <InfoView>
+          {!!ensName && (
+            <TextWithCopy textToCopy={ensName} toastText={t('toast.ensNameCopiedToClipboard')} iconColor={colors.link}>
               {ensName}
-            </BaseText>
-            <BaseText regular center secondary>{t('label.yourEnsName')}</BaseText>
-          </InfoView>
-        )}
-        <QRCodeWrapper>
-          <View style={{ overflow: 'hidden', padding: 10 }}>
-            {!!address && <QRCodeWithTheme value={address} size={160} />}
-          </View>
-          <WalletAddress onPress={() => handleCopyToClipboard(address)}>
-            {address}
-          </WalletAddress>
-        </QRCodeWrapper>
-        <ButtonsRow>
-          {showBuyTokensButton && (
-          <Button
-            title={t('button.buyTokens')}
-            onPress={handleBuyTokens}
-            primarySecond
-            small={needsSmallButtons}
-            style={{ flex: 1, marginRight: 10 }}
-          />
+            </TextWithCopy>
           )}
-          <Button
-            title={t('button.shareAddress')}
-            onPress={handleAddressShare}
-            small={needsSmallButtons}
-            style={{ flex: 1 }}
-          />
-        </ButtonsRow>
-        {showBuyTokensButton && (
-        <IconsContainer>
-          <Image source={visaIcon} />
-          <IconsSpacing />
-          <Image source={mastercardIcon} />
-        </IconsContainer>
-        )}
+          {ensName ? (
+            <WalletAddress>{address}</WalletAddress>
+          ) : (
+            <TextWithCopy
+              toastText={t('toast.addressCopiedToClipboard')}
+              textToCopy={address}
+              textStyle={{ color: colors.basic030 }}
+              iconColor={colors.link}
+            >
+              {address}
+            </TextWithCopy>
+          )}
+        </InfoView>
+        <QRCodeWrapper>{!!address && <QRCodeWithTheme value={address} size={104} />}</QRCodeWrapper>
```

Comment:
`QRCodeWrapper` has margin, I guess it would be better to not show it as well if there's no address?
```suggestion
        {!!address && <QRCodeWrapper><QRCodeWithTheme value={address} size={104} /></QRCodeWrapper>}
```

Reply:
Done.